assets/js/blocks/stock-filter/block.js

```diff
@@ -183,6 +233,35 @@ const StockStatusFilterBlock = ( {
 		}
 	}, [ checked, currentCheckedQuery, previousCheckedQuery ] );
 
+	/**
+	 * Important: For PHP rendered block templates only.
+	 */
+	useEffect( () => {
+		if ( isRenderingPHPTemplate ) {
+			setChecked( checked );
+			// Only automatically redirect if the filter button is not active.
+			if ( ! blockAttributes.showFilterButton ) {
+				redirectPageForPhpTemplate( checked );
+			} else {
+				setProductStockStatusQuery(
+					getActiveFilters( STOCK_STATUS_OPTIONS, QUERY_PARAM_KEY )
+				);
+			}
```

Comment:
This code will not only set the active filter on load but also force the stock status filter to follow the value in the initial URL. With this, the stock filter can't be removed from the Active Filters block (while it's possible to remove it when the filter button is disabled):

https://user-images.githubusercontent.com/5423135/164211007-b4d55a57-9729-49b8-8ba7-b607ad539e9b.mov